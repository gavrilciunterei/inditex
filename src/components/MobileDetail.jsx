import React from 'react';
import { useTranslation } from 'react-i18next';
import DescriptionItems from './DescriptionItems';

function MobileDetail({ data }) {
  const {
    brand,
    model,
    price,
    cpu,
    ram,
    os,
    displayResolution,
    battery,
    primaryCamera,
    secondaryCmera,
    dimentions,
    weight,
  } = data;

  const { t } = useTranslation();

  return (
    <div>
      <DescriptionItems name={t('detail.brand')} text={brand} />
      <DescriptionItems name={t('detail.model')} text={model} />
      <DescriptionItems name={t('detail.price')} text={price} />
      <DescriptionItems name={t('detail.cpu')} text={cpu} />
      <DescriptionItems name={t('detail.ram')} text={ram} />
      <DescriptionItems name={t('detail.os')} text={os} />
      <DescriptionItems
        name={t('detail.displayResolution')}
        text={displayResolution}
      />
      <DescriptionItems name={t('detail.battery')} text={battery} />
      <DescriptionItems name={t('detail.primaryCamera')} text={primaryCamera} />
      <DescriptionItems
        name={t('detail.secondaryCmera')}
        text={secondaryCmera}
      />
      <DescriptionItems name={t('detail.dimentions')} text={dimentions} />
      <DescriptionItems name={t('detail.weight')} text={weight} />
    </div>
  );
}

export default MobileDetail;
