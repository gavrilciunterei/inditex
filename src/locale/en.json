{
  "general": {
    "language_es": "Spanish",
    "language_en": "English"
  },
  "home": {
    "search": "Search"
  },
  "detail": {
    "brand": "Brand",
    "model": "Model",
    "price": "Price",
    "cpu": "CPU",
    "ram": "RAM",
    "os": "OS",
    "displayResolution": "Display Resolution",
    "battery": "Battery",
    "primaryCamera": "Primary Camera",
    "secondaryCmera": "Secondary Camera",
    "dimentions": "Dimentions",
    "weight": "Weight",
    "storage": "Storage",
    "color": "Color"
  }
}
