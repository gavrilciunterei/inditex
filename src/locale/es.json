{
  "general": {
    "language_es": "Español",
    "language_en": "Inglés"
  },
  "home": {
    "search": "Buscar",
    "notAvailable": "No disponible"
  },
  "detail": {
    "brand": "Marca",
    "model": "Modelo",
    "price": "Precio",
    "cpu": "CPU",
    "ram": "RAM",
    "os": "Sistema Operativo",
    "displayResolution": "Resolución de pantalla",
    "battery": "Batería",
    "primaryCamera": "Cámara Primaria",
    "secondaryCmera": "Cámara Secundaria",
    "dimentions": "Dimensiones",
    "weight": "Peso",
    "storage": "Almacenamiento",
    "color": "Color",
    "add": "Añadir"
  }
}
