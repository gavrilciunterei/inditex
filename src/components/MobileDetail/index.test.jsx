import React from 'react';
import renderer from 'react-test-renderer';
import MobileDetail from './index';
import '../../locale/index';

const data = {
  brand: 'Acer',
  model: 'Iconia Talk S',
  price: '160',
  cpu: 'Mali-T720MP2',
  ram: '2 GB RAM',
  os: 'Android',
  displayResolution: '7.0 inches (~69.8% screen-to-body ratio)',
  battery: 'Non-removable Li-Ion 3400 mAh battery (12.92 Wh)',
  primaryCamera: '13 MP, autofocus',
  secondaryCmera: '2 MP, 720p',
  dimentions: '191.7 x 101 x 9.4 mm (7.55 x 3.98 x 0.37 in)',
  weight: '260',
};

it('Test MobileDetail', () => {
  const tree = renderer.create(<MobileDetail data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
