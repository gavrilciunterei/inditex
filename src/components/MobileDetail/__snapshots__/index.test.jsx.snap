// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MobileDetail 1`] = `
<div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Brand
        :
      </p>
      <p>
        Acer
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Model
        :
      </p>
      <p>
        Iconia Talk S
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Price
        :
      </p>
      <p>
        160
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        CPU
        :
      </p>
      <p>
        Mali-T720MP2
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        RAM
        :
      </p>
      <p>
        2 GB RAM
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        OS
        :
      </p>
      <p>
        Android
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Display Resolution
        :
      </p>
      <p>
        7.0 inches (~69.8% screen-to-body ratio)
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Battery
        :
      </p>
      <p>
        Non-removable Li-Ion 3400 mAh battery (12.92 Wh)
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Primary Camera
        :
      </p>
      <p>
        13 MP, autofocus
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Secondary Camera
        :
      </p>
      <p>
        2 MP, 720p
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Dimentions
        :
      </p>
      <p>
        191.7 x 101 x 9.4 mm (7.55 x 3.98 x 0.37 in)
      </p>
    </div>
  </div>
  <div>
    <div
      className="inline-flex"
    >
      <p
        className="font-bold mr-2"
      >
        Weight
        :
      </p>
      <p>
        260
      </p>
    </div>
  </div>
</div>
`;
